packages/odf/components/kms-config/azurekms-connection-form.tsx

```diff
@@ -0,0 +1,179 @@
+import * as React from 'react';
+import { FieldLevelHelp } from '@odf/shared/generic/FieldLevelHelp';
+import { useCustomTranslation } from '@odf/shared/useCustomTranslationHook';
+import { getValidatedProp } from '@odf/shared/utils';
+import { Trans } from 'react-i18next';
+import { FormGroup, TextInput } from '@patternfly/react-core';
+import { validateConnectionName } from '../../constants';
+import { VaultConfig, ThalesConfig, AzureConfig } from '../../types';
+import { isValidName } from './utils';
+
+export const AzureKMSConnectionForm: React.FC<AzureKMSConnectionFormProps> = ({
+  className,
+  kmsState,
```

Comment:
`kmsState` for all reads and `kmsStateClone` for all write/updates...
```suggestion
  kmsState,
  kmsStateClone,
```

Reply:
Ok